subsys/bluetooth/host/hci_core.c

```diff
@@ -93,6 +93,11 @@ static struct bt_pub_key_cb *pub_key_cb;
 static bt_dh_key_cb_t dh_key_cb;
 #endif /* CONFIG_BT_ECC */
 
+#if defined(CONFIG_BT_CONN)
+static bt_remote_version_cb *remote_version_cb;
+K_SEM_DEFINE(remote_version_sem, 1, 1);
```

Comment:
Why do you need a semaphore?

Reply:
@Vudentz wanted it in case it was called several times in a short timeframe

Reply:
Hmm, I guess it'll deadlock then if it's called a second time from the RX thread (say, a connected callback)? Sorry that this PR has dragged on for so long, but I'd like to discuss some other options, such as making the event a priority one (i.e. having it processed through the priority RX thread). Maybe we could have a short chat with you and @Vudentz on Slack today?

Reply:
Sure, I'm available from about 13:00 (GMT +2), just tell me when.

Reply:
> Sure, I'm available from about 13:00 (GMT +2), just tell me when.

I've pinged you on #bluetooth earlier today. Are you available now @Blackladder ?